src/evict/evict_lru.c

```diff
@@ -631,7 +631,7 @@ __evict_pass(WT_SESSION_IMPL *session)
 		 * of whether the cache is full, to prevent the oldest ID
 		 * falling too far behind.
 		 */
-		WT_RET(__wt_txn_update_oldest(session, false));
+		WT_RET(__wt_txn_update_oldest(session, true, false));
```

Comment:
This is the key fix -- the eviction server should update the oldest transaction ID any time it has fallen behind, but not wait for locks on the snapshot table.